### PR TITLE
[D]Network Peers

### PR DESCRIPTION
**Pull request by michaelkad**
Add [D]: pi_network_peers
Update [D]: pi_network, pi_networks
Update [R]: pi_network

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
=== RUN   TestAccIBMPINetworkPeerOnPrem
--- PASS: TestAccIBMPINetworkPeerOnPrem (43.32s)
PASS

=== RUN   TestAccIBMPINetworksDataSource_basic
--- PASS: TestAccIBMPINetworksDataSource_basic (16.10s)
PASS

=== RUN   TestAccIBMPINetworkPeersDataSourceBasic
--- PASS: TestAccIBMPINetworkPeersDataSourceBasic (15.47s)
PASS

...
```

### CODE REVIEW

File: ibm/service/power/data_source_ibm_pi_network.go [hunk 1]
```diff
@@ -37,7 +37,8 @@ func DataSourceIBMPINetwork() *schema.Resource {
 			// Attributes
 			Attr_AccessConfig: {
 				Computed:    true,
-				Description: "The network communication configuration option of the network (for satellite locations only).",
+				Deprecated:  "This field is deprecated please use peer_id instead.",
+				Description: "The network communication configuration option of the network (for on prem locations only).  Use `peer_id` instead.",
```
Comment (Alexander-Kita) [inline ibm/service/power/data_source_ibm_pi_network.go:41]:
```suggestion
				Description: "The network communication configuration option of the network (for on prem locations only). Use `peer_id` instead.",
```

**Review by Alexander-Kita — DISMISSED** [review]

File: website/docs/d/pi_networks.html.markdown [hunk 2]
```diff
@@ -7,22 +7,26 @@ description: |-
...
 ## Example usage
+
 ```terraform
 data "ibm_pi_networks" "ds_network" {
   pi_cloud_instance_id = "49fba6c9-23f8-40bc-9899-aca322ee7d5b"
 }
 ```
 
-**Notes**
+## Notes
```
Comment (ismirlia) [inline website/docs/d/pi_networks.html.markdown:21]:
```suggestion
### Notes
```

**Review by ismirlia — APPROVED** [review]

File: ibm/service/power/data_source_ibm_pi_network_peers.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+	"github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/power/models"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+)
+
+func DataSourceIBMPINetworkPeers() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIBMPINetworkPeersRead,
+
```
Comment (Alexander-Kita) [inline ibm/service/power/data_source_ibm_pi_network_peers.go:22]:
```suggestion
```

File: ibm/service/power/resource_ibm_pi_network.go [hunk 4]
```diff
@@ -94,9 +94,47 @@ func ResourceIBMPINetwork() *schema.Resource {
...
 				ValidateFunc: validate.ValidateAllowedStringValues([]string{"internal-only", "outbound-only", "bidirectional-static-route", "bidirectional-bgp", "bidirectional-l2out"}),
 				Description:  "PI network communication configuration",
 			},
+			Arg_NetworkPeer: {
+				Description: "Network peer information.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						Attr_ID: {
+							Description: "ID of the network peer.",
+							Optional:    true,
```
Comment (Alexander-Kita) [inline ibm/service/power/resource_ibm_pi_network.go:107]:
```suggestion
							Required:    true,
```

File: ibm/service/power/resource_ibm_pi_network.go [hunk 5]
```diff
@@ -529,3 +602,35 @@ func isPERWorkspaceRefreshFunc(client *instance.IBMPIWorkspacesClient, id string
...
 	}
 }
+func networkMapToNetworkCreatePeer(networkCreatePeerMap map[string]interface{}) (*models.NetworkCreatePeer, error) {
+	ncp := &models.NetworkCreatePeer{}
+	if networkCreatePeerMap[Attr_ID].(string) != "" {
+		id := networkCreatePeerMap[Attr_ID].(string)
+		ncp.ID = &id
+	} else {
+		return nil, fmt.Errorf("%s is required with %s", Attr_ID, Arg_NetworkPeer)
+	}
```
Comment (Alexander-Kita) [inline ibm/service/power/resource_ibm_pi_network.go:612]: Can remove this if statement if schema is updated with required

**Review by Alexander-Kita — DISMISSED** [review]

**Review by ismirlia — APPROVED** [review]

**Comment by michaelkad** [comment]
```
=== RUN   TestAccIBMPINetworkPeersDataSourceBasic
--- PASS: TestAccIBMPINetworkPeersDataSourceBasic (19.79s)
PASS

=== RUN   TestAccIBMPINetworksDataSource_basic
--- PASS: TestAccIBMPINetworksDataSource_basic (16.51s)
PASS

=== RUN   TestAccIBMPINetworkPeerOnPrem
--- PASS: TestAccIBMPINetworkPeerOnPrem (45.62s)
PASS

```

**Review by ismirlia — DISMISSED** [review]

File: ibm/service/power/ibm_pi_constants.go [hunk 6]
```diff
@@ -279,9 +280,11 @@ const (
 	Attr_Name                               = "name"
 	Attr_NetworkAddressGroupID              = "network_address_group_id"
 	Attr_NetworkAddressGroups               = "network_address_groups"
+	Attr_NetworkAddressTranslation                   = "network_address_translation"
```
Comment (Alexander-Kita) [inline ibm/service/power/ibm_pi_constants.go:283]: Spacing needs to be fixed on these

Comment (michaelkad) [reply ibm/service/power/ibm_pi_constants.go]: Done

**Review by Alexander-Kita — APPROVED** [review]

**Comment by Alexander-Kita** [comment]
Opened in main repo: https://github.com/IBM-Cloud/terraform-provider-ibm/pull/5835

**Pull request closed without merging**